registry.go

```diff
@@ -87,14 +88,16 @@ func (r *Registry) SetLogger(logger Logger) {
 	r.config.Log = logger
 }
 
-func (r *Registry) Start() {
+func (r *Registry) Start() error {
 	if r.config == nil || r.config.Uri == "" {
-		r.config.Log.Infof("Registry config has no uri. Ignoring Start request.")
-		return
+		err := fmt.Errorf("registry config has no uri. Ignoring Start request")
+		r.config.Log.Infof(err)
```

Comment:
This doesn't work. It needs to be something like `r.config.Log.Infof(err.Error())` or `...Infof("%v", err)`

Reply:
I guess I trusted my IDE too much on that one. I updated it to `fmt.Sprintf`.

Reply:
I'd prefer if it remained an `Error`, otherwise returning a string to indicate an error seems a bit counter-intuitive.